packages/vhd-lib/src/Vhd/VhdSynthetic.integ.spec.js

```diff
@@ -0,0 +1,64 @@
+/* eslint-env jest */
+
+import rimraf from 'rimraf'
+import tmp from 'tmp'
+import { Disposable, pFromCallback } from 'promise-toolbox'
+import { getSyncedHandler } from '@xen-orchestra/fs'
+
+import { SECTOR_SIZE } from '../_constants'
+import { createRandomFile, convertFromRawToVhd } from '../tests/utils'
+import { openVhd } from '../../dist'
+import { VhdSynthetic } from './VhdSynthetic'
+
+let tempDir = null
+
+jest.setTimeout(60000)
+
+beforeEach(async () => {
+  tempDir = await pFromCallback(cb => tmp.dir(cb))
+})
+
+afterEach(async () => {
+  await pFromCallback(cb => rimraf(tempDir, cb))
+})
+
+test('It can read block from a synthetic vhd', async () => {
+  const bigRawFileName = `${tempDir}/bigrandomfile`
+  await createRandomFile(bigRawFileName, 8)
+  const bigVhdFileName = `${tempDir}/bigrandomfile.vhd`
+  await convertFromRawToVhd(bigRawFileName, bigVhdFileName)
+
+  const smallRawFileName = `${tempDir}/smallrandomfile`
+  await createRandomFile(smallRawFileName, 4)
+  const smallVhdFileName = `${tempDir}/smallrandomfile.vhd`
+  await convertFromRawToVhd(smallRawFileName, smallVhdFileName)
+
+  await Disposable.use(async function* () {
+    const handler = yield getSyncedHandler({ url: 'file:///' })
+    const smallVhd = yield openVhd(handler, smallVhdFileName)
+    const bigVhd = yield openVhd(handler, bigVhdFileName)
+    const syntheticVhd = new VhdSynthetic([smallVhd, bigVhd])
```

Comment:
`Disposable.all` is exactly here for this purpose :slightly_smiling_face: 
```suggestion
    const syntheticVhd = new VhdSynthetic(yield Disposable.all([
      openVhd(handler, smallVhdFileName),
      openVhd(handler, bigVhdFileName)
    ]))
```